abcclassroom/dirsetup.py

```diff
@@ -0,0 +1,94 @@
+"""
+abc-classroom.dir-setup
+========================
+
+"""
+
+import os
+import shutil
+import argparse
+import abcclassroom
+
+
+def path_to_example(dataset):
+    """
+    Gets the path to an example dataset.
+
+    Parameters
+    ----------
+    dataset : string
+        stringy boi
+    """
+    abcclassroom_path = os.path.split(abcclassroom.__file__)[0]
+    data_dir = os.path.join(abcclassroom_path, "example-data")
+    data_files = os.listdir(data_dir)
+    if dataset not in data_files:
+        raise KeyError(dataset + " not found in abc-classroom example data.")
+    return os.path.join(data_dir, dataset)
+
+
+def directory_setup():
+    """
+    Construct a file path to an example dataset.
+    This file defines helper functions to access data files in this directory,
+    to support examples. Adapted from the PySAL package.
+
+    Parameters
+    ----------
+    dataset: string
+        Name of a dataset to access (e.g., "epsg.json", or "RGB.byte.tif")
+
+    Returns
+    -------
+        A file path (string) to the dataset
+    """
+    # Creating the default names for the cloned and template repos.
+    cloned = "cloned_repos"
+    template = "template_repos"
+    course = "course_dir"
+    # Making sure the configuration file is where it's supposed to be.
+    config = path_to_example("sample_config.yml")
+    if not os.path.exists(config):
+        raise ValueError(
+            "Configuration file can't be located, please ensure abc-classroom has been installed correctly"
+        )
+    # Allows users to rename the cloned and template repos.
+    parser = argparse.ArgumentParser(description=directory_setup.__doc__)
+    parser.add_argument(
+        "--course_repo",
+        help="Name of the main course repository"
+    )
+    parser.add_argument(
+        "--cloned_repo",
+        help="Name of the repository to hold the cloned files"
+    )
+    parser.add_argument(
+        "--template_repo",
+        help="Name of the repository to hold the template files"
+    )
+    parser.add_argument(
+        "--override_existing",
+        help="Option to override the existing folder structure made by this function previously."
+    )
+    args = parser.parse_args()
+    # Assigning the custom folder names if applicable
+    if args.cloned_repo:
+        cloned = args.cloned_repo
+    if args.template_repo:
+        template = args.template_repo
+    if args.course_repo:
+        course = args.course_repo
+    main_dir = os.path.join(os.getcwd(), course)
+    if args.override_existing == "True" and os.path.isdir(main_dir):
+        shutil.rmtree(main_dir)
```

Comment:
I am always a bit scared of `rm` in scripts. Maybe better to return an error that the dir exists and ask the user to either pick a new name or rename / delete the existing directory?

Reply:
totally. i thought that we were going to provide flags for 
1. merge `-m` and
2. overwrite `-f` or something like that via our slack conversation. 

If we overwrite then... well i would imagine we would clean house. are you suggesting we don't offer an overwrite argument? 

Here is the slack conversation
```
1. if the directory exists, raise an error and ask for a flag (what to do)
flag options:
-m merge: overwrite any files that exist with the same name in that directory and add any new ones. leave whatever is in there already alone!
-d  delete: delete that directory and start fresh (edited) 
do those two flags sound reasonable? choose one or the other
```

Reply:
lemme know if i misunderstood how this would work but that was my take away that users can force a delete if they want.

Reply:
This all sounds good to me - sorry for the confusion!

Reply:
ok cool. so you are comfortable with the rm IF the user uses the `-f` flag?